daft/datatype.py

```diff
@@ -249,6 +249,9 @@ def map(cls, key_type: DataType, value_type: DataType) -> DataType:
     def struct(cls, fields: dict[str, DataType]) -> DataType:
         """Create a Struct DataType: a nested type which has names mapped to child types
 
+        Example:
```

Comment:
another api that might be nice could be 
```
DataType.struct(name=DataType.string(), age=DataType.int64())
```

Reply:
Yeah could be nice. Do we foresee that we might want to leverage the `kwargs` for anything else? Otherwise I can go ahead and add that